src/services/job-processor/service.ts

```diff
@@ -68,56 +73,60 @@ export const makeJobProcessor = ({
     await executionApi.resolveExitBusAddress()
     await executionApi.resolveConsensusAddress()
 
-    const toBlock = await executionApi.latestBlockNumber()
-    const fromBlock = toBlock - eventsNumber
-    logger.info('Fetched the latest block from EL', { latestBlock: toBlock })
-
-    for (const operatorId of operatorIds) {
-      logger.info('Fetching request events from the Exit Bus', {
-        eventsNumber,
-        fromBlock,
-        toBlock,
-      })
-
-      const eventsForEject = await executionApi.logs(
-        fromBlock,
-        toBlock,
-        operatorId
-      )
-
-      logger.info('Handling ejection requests', {
-        amount: eventsForEject.length,
-      })
-
-      for (const [ix, event] of eventsForEject.entries()) {
-        logger.debug(`Handling exit ${ix + 1}/${eventsForEject.length}`, event)
-
-        try {
-          if (await consensusApi.isExiting(event.validatorPubkey)) {
-            logger.info('Validator is already exiting(ed), skipping')
-            continue
-          }
+    const lastBlockNumber = await executionApi.latestBlockNumber()
 
-          if (config.DRY_RUN) {
-            logger.info('Not initiating an exit in dry run mode')
-            continue
-          }
+    const eventsForEject = await exitLogs.getLogs(operatorIds, lastBlockNumber)
 
-          if (config.VALIDATOR_EXIT_WEBHOOK) {
-            await webhookProcessor.send(config.VALIDATOR_EXIT_WEBHOOK, event)
-          } else {
-            await messagesProcessor.exit(messageStorage, event)
+    logger.info('Handling ejection requests', {
+      amount: eventsForEject.length,
+    })
+
+    for (const [ix, event] of eventsForEject.entries()) {
+      logger.debug(`Handling exit ${ix + 1}/${eventsForEject.length}`, event)
+
+      if (event.acknowledged) {
+        logger.debug('Event already acknowledged, skipping')
+        continue
+      }
+
+      try {
+        if (await consensusApi.isExiting(event.validatorPubkey)) {
+          logger.info('Validator is already exiting(ed), skipping', {
+            validatorIndex: event.validatorIndex,
+          })
+          // Acknowledge the event to avoid processing it again
+          // We do an additional check, because if we didn't check for finalized,
+          // we might miss the reorganization.
+          if (
+            lastBlockNumber - event.blockNumber > FINALIZED_BLOCK_EQUIVALENT ||
+            (await consensusApi.isExiting(event.validatorPubkey, 'finalized'))
+          ) {
+            event.ack()
           }
-        } catch (e) {
-          logger.error(`Unable to process exit for ${event.validatorPubkey}`, e)
-          metrics.exitActions.inc({ result: 'error' })
+          continue
+        }
+
+        if (config.DRY_RUN) {
+          logger.info('Not initiating an exit in dry run mode')
+          continue
+        }
+
+        if (config.VALIDATOR_EXIT_WEBHOOK) {
```

Comment:
Maybe we should try to exit if some checks below failed?
eg `consensusApi.isExiting` raises error

Reply:
We have a mechanism for additional attempts when sending requests, and in case of an error, we will restart the loop after 10 seconds